src/coreclr/vm/siginfo.cpp

```diff
@@ -386,6 +386,26 @@ void SigPointer::ConvertToInternalSignature(Module* pSigModule, SigTypeContext *
         cArgs--;
     }
 }
+
+void SigPointer::CopySignature(Module* pSigModule, SigBuilder* pSigBuilder, SigPointer* pSigPtrTokenEnd, BYTE additionalCallConv)
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        STANDARD_VM_CHECK;
+    }
+    CONTRACTL_END
+
+    BYTE element;
+    IfFailThrowBF(GetByte(&element), BFA_BAD_COMPLUS_SIG, pSigModule);
+    pSigBuilder->AppendByte(element | additionalCallConv);
+
+    while (m_ptr != pSigPtrTokenEnd->m_ptr)
+    {
+        IfFailThrowBF(GetByte(&element), BFA_BAD_COMPLUS_SIG, pSigModule);
+        pSigBuilder->AppendByte(element);
+    }
```

Comment:
```suggestion
    pSigBuilder->AppendBlob(m_ptr, pSigPtrTokenEnd->m_ptr - m_ptr);
```
Once you do this, it is not clear whether it is worth it to have this as a method on SigPointer. It is very specialized small method. I think the 4 lines can be inline in the one place where this is used.

Reply:
I did the main suggestion of AppendBlob, but keeping it as a method prevents having to use discouraged `SigParser::GetPtr()` to get at the protected `m_ptr`